### PR TITLE
New Installation Screen

### PR DESCRIPTION
**Pull request by FinalAngel**
1. **This PR will replace https://github.com/divio/django-cms/pull/4326**
2. **Add page will be replaced with the new wizard once it's merged**

**Step 1**
image

you get automatically redirected to the admin: ´´/en/admin/login/?next=/en/´´. The screen above is shown when you have javascript disabled.

**Step 2**
image

after login, you'll get back to the previous screen, the modal will open automatically:

image

### CODE REVIEW

File: cms/templates/cms/welcome.html [hunk 1]
```diff
@@ -1,74 +1,89 @@
...
-        var modal = new CMS.Modal();
-            modal.open({
-                url: '{% cms_admin_url "cms_page_add" %}?title=Home'
-        });
-    });
-}
-</script>
+    {% render_block "js" %}
 
+    <script>
```
Comment (FinalAngel) [inline cms/templates/cms/welcome.html]: might not be the best to use inline js here, but would be a waste to just implement another file?

**Comment by vxsx** [comment]
after you login, if you close the modal - can you somehow reopen it or would you need to refresh?

File: cms/static/cms/sass/components/_welcome.scss [hunk 2]
```diff
@@ -0,0 +1,91 @@
+//##############################################################################
+// #WELCOME
+//
+// Shown when there is no page available
+.cms-welcome-bg {
+    background: #f7f7f7;
```
Comment (vxsx) [inline cms/static/cms/sass/components/_welcome.scss]: maybe reuse some of the gray vars here?

File: cms/static/cms/sass/components/_welcome.scss [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        text-align: center;
+        padding: 40px 0 15px;
+        border-bottom: 1px solid lighten($gray-lighter, 10%);
+        .cms-icon {
+            color: $color-success;
+        }
+    }
+
+    .cms-welcome-logo {
+        @extend .cms-icon;
```
Comment (vxsx) [inline cms/static/cms/sass/components/_welcome.scss]:
```
@extend %icon;
@include icon(logo);
```

File: cms/static/cms/sass/components/_welcome.scss [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    // section area
+    .cms-welcome-section {
+        box-sizing: border-box;
+        box-shadow: 0 0 5px $gray-lighter;
+        padding: 20px;
+        margin-bottom: 5px;
+        background: white;
+        h2 {
+            font-size: 16px;
```
Comment (vxsx) [inline cms/static/cms/sass/components/_welcome.scss]: should it come from variable?

File: cms/static/cms/sass/components/_welcome.scss [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        @extend .cms-icon-logo;
+        font-size: 35px;
+        margin: 50px auto 30px;
+    }
+
+    // section area
+    .cms-welcome-section {
+        box-sizing: border-box;
+        box-shadow: 0 0 5px $gray-lighter;
+        padding: 20px;
```
Comment (vxsx) [inline cms/static/cms/sass/components/_welcome.scss]: should it come from the variable?

File: cms/static/cms/sass/components/_welcome.scss [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        margin-bottom: 5px;
+        background: white;
```
Comment (vxsx) [inline cms/static/cms/sass/components/_welcome.scss]: should it come from the variable?

File: cms/static/cms/sass/components/_welcome.scss (hunk 2 above)
Comment (FinalAngel) [inline cms/static/cms/sass/components/_welcome.scss]: used it everywhere, this is special :/

File: cms/templates/cms/welcome.html [hunk 7]
```diff
@@ -1,74 +1,89 @@
...
+                {% blocktrans %}
+                    Welcome to django CMS version <strong>{{ cms_version }}</strong>.
+                {% endblocktrans %}
+                </p>
+            <p>
+                {% url "admin:index" as admin_index %}
+                {% url "admin:cms_page_add" as admin_add_page %}
+                {% blocktrans %}
+                    You can also login through the <a href="{{ admin_index }}">admin</a> and
+                    <a href="{{ admin_add_page }}?title=Home" class="js-welcome-add">add a page</a> manually
```
Comment (vxsx) [inline cms/templates/cms/welcome.html]: why the "Javascript is disabled" ? if javascript is disabled user is really out of luck, you can't really use cms

File: cms/templates/cms/welcome.html [hunk 8]
```diff
@@ -1,74 +1,89 @@
...
-                url: '{% cms_admin_url "cms_page_add" %}?title=Home'
-        });
-    });
-}
-</script>
+    {% render_block "js" %}
 
+    <script>
+        {% if user.is_authenticated %}
+            var btn = CMS.$('.js-welcome-add');
```
Comment (vxsx) [inline cms/templates/cms/welcome.html]: where's the document ready?

File: cms/templates/cms/welcome.html [hunk 9]
```diff
@@ -1,74 +1,89 @@
...
+            btn.on('click', function (e) {
+                e.preventDefault();
+                var modal = new CMS.Modal();
+                modal.open({
+                    url: '{% cms_admin_url "cms_page_add" %}?title=Home'
+                });
+            });
+            // CMS.Modal is not yet ready, we need to wait a bit and then
```
Comment (vxsx) [inline cms/templates/cms/welcome.html]: should be ready to use when document is ready

Comment (FinalAngel) [inline cms/templates/cms/welcome.html]: nope

File: cms/static/cms/sass/components/_welcome.scss [hunk 10]
```diff
@@ -0,0 +1,94 @@
+//##############################################################################
+// #WELCOME
+//
+// Shown when there is no page available
+$welcome-width: 60%;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_welcome.scss]: should we also move it to general settings?

**Comment by vxsx** [comment]
:+1:

**Comment by mkoistinen** [comment]
:+1:

**Pull request merged**